scripts/workers/submission_worker.py

```diff
@@ -574,8 +577,7 @@ def run_submission(
         submission.submission_metadata_file.save(
             "submission_metadata.json", ContentFile(submission_metadata)
         )
-
-    submission.save()
```

Comment:
Why are we changing this line to `if` block? How is it causing redundancy?

Reply:
we need above save only if `submission_output != None` see this https://github.com/Cloud-CV/EvalAI/pull/3090/files#diff-ae6210f157b1ef66d873b08fe30f65f36c92c1e3d8aa61e3c8580910f5f06695L560. For a failed submission we are calling save twice.